regional_mom6/utils.py

```
@@ -384,3 +385,23 @@ def is_rectilinear_hgrid(hgrid: xr.Dataset, rtol: float = 1e-3) -> bool:
     ):
         return True
     return False
+
+
+def find_files_by_pattern(paths: list, patterns: list, error_message=None) -> list:
+    """
+    Function searchs paths for patterns and returns the list of the file paths with that pattern
+    """
+    # Use glob to find all files
+    all_files = []
+    for pattern in patterns:
+        for path in paths:
+            all_files.extend(Path(path).glob(pattern))
+
+    if len(all_files) == 0:
+        if error_message is None:
+            return "No files found at the following paths: {} for the following patterns: {}".format(
+                paths, patterns
+            )
+        else:
+            return error_message
```

Seems like this should raise an exception rather than breaking type contract and returning a string?

---

Yup! Thanks